tensorflow/java/src/gen/java/org/tensorflow/processor/operator/OperatorProcessor.java

```diff
@@ -0,0 +1,115 @@
+
+  @Override
+  public SourceVersion getSupportedSourceVersion() {
+    return SourceVersion.latestSupported();
+  }
+
+  @Override
+  public synchronized void init(ProcessingEnvironment processingEnv) {
+    super.init(processingEnv);
+    messager = processingEnv.getMessager();
+    filer = processingEnv.getFiler();
+  }
+
+  @Override
+  public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+    // Nothing needs to be done at the end of all rounds.
+    if (roundEnv.processingOver()) {
+      return false;
+    }
+
+    // Collect all classes tagged with our annotation.
+    Set<TypeElement> opClasses = new HashSet<TypeElement>();
+    for (TypeElement annotation : annotations) {
+      if (!collectOpClasses(roundEnv, opClasses, annotation)) {
+        return true;
+      }
+    }
+
+    // Nothing to do when there are no tagged classes.
+    if (opClasses.isEmpty()) {
+      return false;
+    }
+
+    // TODO:(kbsriram) validate operator classes and generate Op API.
+    writeApi();
+    hasRun = true;
+    return true;
+  }
+
+  @Override
+  public Set<String> getSupportedAnnotationTypes() {
+    return Collections.singleton(String.format("%s.annotation.Operator", OP_PACKAGE));
+  }
+
+  private void writeApi() {
+    // Generate an empty class for now and get the build working correctly. This will be changed to
+    // generate the actual API once we've done with build-related changes.
+    // TODO:(kbsriram)
+    try (PrintWriter writer =
+        new PrintWriter(filer.createSourceFile(String.format("%s.Ops", OP_PACKAGE)).openWriter())) {
+      writer.println(String.format("package %s;", OP_PACKAGE));
+      writer.println("public class Ops{}");
+    } catch (IOException e) {
+      error(null, "Unexpected failure generating API: %s", e.getMessage());
+    }
+  }
+
+  private boolean collectOpClasses(
+      RoundEnvironment roundEnv, Set<TypeElement> opClasses, TypeElement annotation) {
+    boolean result = true;
+    for (Element e : roundEnv.getElementsAnnotatedWith(annotation)) {
+      // This processor has to aggregate all op classes in one round, as it generates a single Ops
+      // API class which cannot be modified once generated. If we find an annotation after we've
+      // generated our code, flag the location of each such class.
+      if (hasRun) {
+        error(e, "Cannot handle generated @Operators");
+        result = false;
+        continue;
+      }
+
+      // @Operator can only apply to types, so e must be a TypeElement.
+      if (!(e instanceof TypeElement)) {
+        error(e, "@Operator can only be applied to classes");
+        result = false;
+        continue;
+      }
+      opClasses.add((TypeElement) e);
+    }
+    return result;
+  }
+
+  private void error(Element e, String message, Object... args) {
+    message(Kind.ERROR, e, message, args);
+  }
+
+  private void message(Kind kind, Element e, String msg, Object... args) {
```

Comment:
Optional: If `message` is used only for `error`, perhaps inline it in `error`?

Reply:
done